src/accesspattern/arrays.jl

```diff
@@ -0,0 +1,24 @@
+typealias NativeArray Union{Array,SubArray}
```

Comment:
Yea, I agree it is sub optimal. And I dislike the name, although it is just internal. I am unsure how to best include all fitting types, but I rather be too inclusive than too loose (example Sparse Arrays). `DataSubset` shouldn't slow anything down really